File: src/dotnet/Ansight.Tools.VisualTree/RemoveOverlayTool.cs
namespace Ansight.Tools.VisualTree;

public sealed class RemoveOverlayTool : ITool
{
    public string Category => "ui";

    public ToolScope Scope => ToolScope.Write;

    public string Id => VisualTreeToolIds.RemoveOverlay;

    public string Name => "Remove Overlay";

    public string Description => "Removes a diagnostic overlay from the active app window by id.";

    public string Keywords => "ui overlay highlight remove clear";

    public ToolSchema ArgumentsSchema => VisualTreeToolSchemas.RemoveOverlayArguments;

    public ToolSchema ResultSchema => VisualTreeToolSchemas.RemoveOverlayResult;

    public ToolSecurity Security => VisualTreeToolSecurityProfiles.RemoveOverlay;

    public Task<ToolResult> Execute(IReadOnlyDictionary<string, string> arguments)
    {
        ArgumentNullException.ThrowIfNull(arguments);
        return VisualTreeSupport.RemoveOverlayAsync(arguments);
    }
}

File: src/ios/Sources/AnsightToolsVisualTree/RemoveOverlayTool.swift
import AnsightCore
import Foundation

public final class RemoveOverlayTool: AnsightTool {
    public init() {}

    public var descriptor: AnsightToolDescriptor {
        AnsightToolDescriptor(
            id: AnsightVisualTreeToolIds.removeOverlay,
            name: "Remove Overlay",
            description: "Removes a diagnostic overlay from the active app window by id.",
            category: "ui",
            policy: .write,
            keywords: "ui overlay highlight remove clear",
            argumentsSchema: AnsightVisualTreeToolSchemas.removeOverlayArguments,
            resultSchema: AnsightVisualTreeToolSchemas.removeOverlayResult
        )
    }

    public func execute(arguments: [String: String]) throws -> AnsightToolExecutionResult {
        AnsightVisualTreeOverlaySupport.removeOverlay(arguments: arguments)
    }
}
